### ISSUE TITLE
Setup is not recognizing Nox Player latest Version. 

### ISSUE THREAD

**Issue by geekoutar** [tmarenko/mff_auto#28]
I've updaed NOX yesterday and now the the aplication is not recognizing the player. 
![2021-05-26_19-45-28](https://user-images.githubusercontent.com/54755247/119740845-fadc2e80-be5a-11eb-98d1-ef9245a64ebd.png)
Attached a screenshot.

**Comment by geekoutar** [comment]
i've tryed executing as administrator and it didn't fix the issue.

**Comment by tmarenko** [comment]
Please edit your screenshot and hide your in-game username (or remove it entirely). Your profile could be banned by devs if they somehow end up here.
Currently I'm using NoxPlayer 7.0.1.0 so I'll try look into new version soon.

**Comment by geekoutar** [comment]
> Please edit your screenshot and hide your in-game username (or remove it entirely). Your profile could be banned by devs if they somehow end up here.
> Currently I'm using NoxPlayer 7.0.1.0 so I'll try look into new version soon.

Thanks!!

**Comment by tmarenko** [comment]
Fixed in https://github.com/tmarenko/mff_auto/commit/6c43146eef5e3413a226bb013870343219a9e7a1
I'm not planning to make a new release on this week so if you want to use new version of NoxPlayer then apply changes from https://github.com/tmarenko/mff_auto/commit/6c43146eef5e3413a226bb013870343219a9e7a1 manually.

**Comment by geekoutar** [comment]
It recognizes NOX but clicks arent passing through the app to the emulator.

**Comment by tmarenko** [comment]
This should fix it: https://github.com/tmarenko/mff_auto/commit/271001a17b0540cca03da9e20df4fc32cdb1c52e
Still I've encountered another problem on 7.0.1.1 version that key handle window is missing.
NoxPlayer's devs restructured window's handles in 7.0.1.1 apparently so all HWNDs are changed.
I managed to found main HWND (for clicks) but have no luck with HWND for keyboard interaction.
I'll kook into it again later.

**Comment by dbzdivik** [comment]
++ any luck still with Update ?

**Comment by tmarenko** [comment]
No. I haven't tested new versions (7.0.1.2 and 7.0.1.3) because I don't have much time for this project right now but as far I can tell in 7.0.1.1 there is not distinct way to determine HWND for keyboard intercation.

**Comment by tmarenko** [comment]
Another way is force-closing MFF application via `adb` bridge:
https://support.bluestacks.com/hc/en-us/articles/360053245871-Settings-menu-on-BlueStacks-5#%E2%80%9C6%E2%80%9D
https://www.bignox.com/blog/how-to-connect-android-studio-with-nox-app-player-for-android-development-and-debug/

But this would require to implement enabling the debug bridge for each emulator and support for multiple Android instances within emulator.

I'm looking forward for Windows 11's feature for installing Android apps into the system without any additional emulators. That would be a great opportunity to skip support for Nox/BlueStack and do work just within OS itself.

For now NoxPlayer's version will be restricted up to 7.0.1.0: https://github.com/tmarenko/mff_auto/commit/fbcfd128de26c9dd8716e0af26f0f5db916714ce

**Comment by geniusdragon** [comment]
Does this mean it won't be updated in the future at some point to support newer versions? I had to update my Nox to support 64bit and it's updated to 7.0.1.6 - which means it's now not detected.

**Comment by tmarenko** [comment]
Eventually yes. Windows 11 should be released at October this year. I'll check it's features and I'll release either an update with Windows 11 support or an update with Android Debug Bridge support

**Comment by geniusdragon** [comment]
I apologise for bringing this up again - I've recently uninstalled nox (as it upgraded to the latest version when I went to install the 64bit version separately), and reinstalled the 7.0.1.0 version, however after rerunning the setup numerous times, the bot still doesn't pick up nox. I'm running bluestacks for the first time purely to be able to run this at the moment, but I'd rather use nox. I uninstalled the latest version, reinstalled the last supported version, and removed the game.json file to reset the setup. Is there anything I may have missed?

**Comment by tmarenko** [comment]
For those who insterested and willing to test, I've started to adding ADB support at this branch: https://github.com/tmarenko/mff_auto/tree/adb_test

Last build: https://github.com/tmarenko/mff_auto/actions/runs/1351558311

For **NoxPlayer** you simply need to use this menu:

![image](https://user-images.githubusercontent.com/31345208/137628742-fe88adaf-5bfb-45cc-897c-5bb9d3c93b71.png)
Result of setup will be seen in log file:

![image](https://user-images.githubusercontent.com/31345208/137628761-50bf3f91-c7b1-40fc-afd0-182424298bf4.png)

After that just restart bot and ADB should connect itself automatically.

For **BlueStacks** you need to enable ADB in Settings:

![image](https://user-images.githubusercontent.com/31345208/137628821-38f8710b-869d-4a7c-8320-219189a1be88.png)

and manually copy your device's serial (eg 127.0.0.1:5555 from screenshot) to `game.json` file:

![image](https://user-images.githubusercontent.com/31345208/137628881-18d25ad4-d07c-4033-877b-4b9cd79ee6ee.png)

Just in case, close other emulators except the one with MFF installed before doing this.

Basically now you can restart game in any of the emulators (Nox or BlueStacks), touch and swipes are doing over ADB directly. Only screen capturing still using DirectX so bot is still sensetive to emulator's screen resolution.

**Comment by tmarenko** [comment]
> Is there anything I may have missed?

Does it say in the log file that it's right version? `Initialized NoxPlayer object with name NoxPlayer version 7.0.1.0`?
Nox has different behaviour for handles since `7.0.1.1`. The version is parsed from `Nox.exe`:
![image](https://user-images.githubusercontent.com/31345208/137632822-14d3ebfc-4b99-461c-b48a-4c22f7fff4d7.png)

**Comment by geniusdragon** [comment]
Bluestacks was successful and is working quite well :)

**Comment by KnightM1** [comment]
> > 2021-10-19 01:18:16,461 - DEBUG - android_emulator - Found devices over ADB: []
> 
> For some reason no device was found. Usually Nox connects automatically every emulator:
> 
> ![image](https://user-images.githubusercontent.com/31345208/137785325-962e4e30-a7c8-47cc-bfc5-c130a714bad2.png)
> 
> UPD: I've tried 7.0.1.6 and it seems it has problems with ADB. You probably will have better luck with BlueStacks because device's serial can be seen right in the settings.

Есть новые версии с использованием ADB что бы юзать новый нокс?
